sqlglot/dialects/clickhouse.py

```diff
@@ -84,6 +84,39 @@ class Parser(parser.Parser):
 
         LOG_DEFAULTS_TO_LN = True
 
+        def _parse_placeholder(self) -> t.Optional[exp.Expression]:
+            """
+            Parse a placeholder expression like SELECT {abc: UInt32} or FROM {table: Identifier}
+            https://clickhouse.com/docs/en/sql-reference/syntax#defining-and-using-query-parameters
+            """
+            index = self._index
+
+            if self._match(TokenType.L_BRACE):
+                # We might be getting a variable from a SELECT clause, or potentially an identifier from a FROM clause
+                if not self._match_set({TokenType.VAR, TokenType.TABLE}):
+                    return None
+
+                this = self._prev.text
+                # Ensure we've got a colon delimiting the placeholder name and placeholder type
+                if not self._match(TokenType.COLON):
```

Comment:
sqlglot is very permissive so we can make colon optional, get rid of the if and just do self._match(TokenType.COLON)